design/control-data-plane-separation/design.md

```diff
@@ -23,377 +23,118 @@ architecture does not allow the control plane and data plane to scale independen
 
 ### Deployment Architecture
 
 ![Deployment architecture](deployment-architecture.png)
 
 - _Control Plane Deployment_: The control plane is a Kubernetes Deployment with one container running the NGF
-  controller. Initially, the control plane will be limited to a single Pod. Once we add leader election, the control
-  plane will be able to scale. The control plane will perform the same functions as it does today, but instead of
+  controller. The control plane will perform the same functions as it does today, but instead of
   configuring nginx by writing files to a shared volume, it will send the configuration to the agent via gRPC.
 - _Control Plane Service_: Exposes the control plane via a Kubernetes Service of type `ClusterIP`. The data plane will
   use the DNS name of the Service to connect to the control plane.
-- _Data Plane DaemonSet/Deployment_: A user can deploy the data plane as either a DaemonSet or Deployment. The data
+- _Data Plane DaemonSet/Deployment_: The data plane can be deployed as either a DaemonSet or Deployment. The data
   plane contains a single container running both the agent and nginx processes. The agent will download the
   configuration from the control plane over a streaming RPC.
-- _NGINX Service_: Exposes nginx via a Kubernetes Service of type `LoadBalancer .`This is the entry point for the
-  customer’s traffic. Initially, this Service will only expose ports 80 and 443. In the future, if we add support for
-  additional listener ports, this Service will expose all the listener ports. Note that this Service should not expose
-  any of the agent’s ports.
+- _NGINX Service_: Exposes nginx via a Kubernetes Service of type `LoadBalancer .` This is the entry point for the
+  customer’s traffic. Note that this Service should not expose any of the agent’s ports.
+
+#### Further Requirements and Implementation Details
+
+- Both deployments should have read only filesystems.
+- Both deployments should have the minimal permissions required to perform their functions.
+- The nginx deployment should be configurable via the helm chart.
+  - Downside of this is that these options will apply to all nginx instances.
+  - We could introduce a CRD, but where would it attach? We already have NginxProxy which controls dynamic data plane configuration, and this may eventually attach to the Gateway instead of just the GatewayClass. Would a Deployment configuration fit in there, and would it be dynamic? That would require us to completely redeploy nginx if a user changes those settings.
+  - We could start with the helm chart option, and rely on user feedback to see if we need to get more granular.
+  - This could also involve creating a ConfigMap that the control plane consumes on startup and contains all nginx Deployment/Daemonset configuration, including NGINX Plus usage configuration.
+- nginx Service should have configurable labels and annotations via the GatewayInfrastructure field in the Gateway resource.
```

Comment:
According to the spec, these labels and annotations should be applied to any resources created in response to the Gateway. Not just the service.